teps/0091-trusted-resources.md

```diff
@@ -230,71 +215,168 @@ spec:
     resources: {}
 ```
 
-`tekton.dev/signature` is used to store the signature.
+Signed Pipeline
+The sample signed pipeline file looks like this:
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  annotations:
+    tekton.dev/signature: MEQCIHhFC480mv7mh/6dxAlp/mvuvGXanuSghMsT+iBhWxt5AiBDvHv8sfKjJ3Ozrzvp+cjr30AOx1SPQDCcaJjlpAoVrA==
+  name: example-Pipeline
+spec:
+  tasks:
+    - name: example-task
+      taskRef:
+        name: example-task
+```
+
+`tekton.dev/signature` is used to store the signature. The future work could be to store the signature in a separate file along with the resource. Remote resolution should be updated to resolve both the resource and the signature.
 
-The signed Resource can be installed directly on the kubernetes cluster or built as OCI bundle to be stored in the registry.
+The signed Resource can be installed directly on the Kubernetes cluster or built as OCI bundle to be stored in the registry.
 
 ### Verify the Resources
 
-The verification should be done in the Tekton Pipeline's admission webhook. A new configmap should be added to gate the verification
-code.
+The verification should be done in the Tekton Pipeline's reconciler after remote resolution. A new configmap should be added to gate the verification code.
+
+Before the verification, the signature is extracted from the resource. For the public key there are several options we can choose by configuration.
+
+In this TEP we propose to store the public keys on the Kubernetes installation. For example, the keys can be configured in a new CRD and deployed to the cluster. This can help to support multiple keys, validate the spec, and have dedicated RBAC policies, so no other components in the cluster can modify the keys.
```

Comment:
I think a CRD makes sense but we'd want to flesh out what the CRD exactly looks like. I'd like to start with the bare-minimum we need i.e. a list of authorities and maybe the patterns to match resources against.
(Would also be useful to call out why a config map would not work for this)

Reply:
Sure! I will update the example of this CRD to have more details.

There are some reasons CRD may be better for configmap:
1. In CRD we can configure multiple keys easily (like in the authorities list) but for configmap there is no list, and users need to figure out what will be the key value in this map. 
2. With CRD we can support other features, like mapping keys to the resources via ResourcePattern.  
3. Better support for keyless.

But yes it is no wrong to use configmap. I will add this back as the default choice.